lib/ansible/modules/cloud/azure/azure_rm_containerinstance.py

```diff
@@ -0,0 +1,429 @@
+            results["instance_previous_finish_time"] = container.instance_view.previous_state.finish_time
+            results["instance_previous_detail_status"] = container.instance_view.previous_state.detail_status
+        # events (list of ContainerEvent)
+    return results
+
+
+class AzureRMContainerInstance(AzureRMModuleBase):
+    """Configuration class for an Azure RM container instance resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            os_type=dict(
+                type='str',
+                default='linux',
+                choices=['linux', 'windows']
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str',
+            ),
+            ip_address=dict(
+                type='str',
+                default='none',
+                choices=['public', 'none']
+            ),
+            ports=dict(
+                type='list',
+                default=[]
+            ),
+            registry_login_server=dict(
+                type='str',
+                default=None
+            ),
+            registry_username=dict(
+                type='str',
+                default=None
+            ),
+            registry_password=dict(
+                type='str',
+                default=None
+            ),
+            containers=dict(
+                type='list',
+                required=True
+            ),
+            force_update=dict(
+                type='bool',
+                default=False
+            ),
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.tags = None
+        self.state = None
+        self.ip_address = None
+
+        self.containers = None
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+
+        super(AzureRMContainerInstance, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                       supports_check_mode=True,
+                                                       supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        self.mgmt_client = self.get_mgmt_svc_client(ContainerInstanceManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        try:
```

Comment:
I believe you don't need to put this in a try/except anymore, the logic is done in azure_rm_common.

Reply:
@zikalino are you able to update this?